x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialRelatesFunction.java

```diff
@@ -203,13 +203,17 @@ public boolean rightDocValues() {
     /**
      * Produce a map of rules defining combinations of incoming types to the evaluator factory that should be used.
      */
-    protected abstract Map<SpatialEvaluatorFactory.SpatialEvaluatorKey, SpatialEvaluatorFactory<?, ?>> evaluatorRules();
+    abstract Map<SpatialEvaluatorFactory.SpatialEvaluatorKey, SpatialEvaluatorFactory<?, ?>> evaluatorRules();
+
+    public abstract boolean isCommutative();
+
+    public abstract SpatialRelatesFunction invert();
```

Comment:
I don't think we need this - `BinaryScalarFunction` already has `replaceChildren` which should do what you need.

Reply:
Woops, never mind - `replaceChildren` won't let you exchange the left and right doc values, sorry!